contrib/epee/include/net/net_ssl.h

```diff
@@ -133,6 +133,8 @@ namespace net_utils
 	constexpr size_t get_ssl_magic_size() { return 9; }
 	bool is_ssl(const unsigned char *data, size_t len);
 	bool ssl_support_from_string(ssl_support_t &ssl, boost::string_ref s);
+
+	bool create_ssl_certificate(EVP_PKEY *&pkey, X509 *&cert);
```

Comment:
This API signature is inadequate. There should be differences between server certs and client certs, but this API doesn't specify which is being created.

In particular, the subjectName of a server cert SHOULD be its fully qualified domain name. And the subjectAltName extension of the cert MUST have its fully qualified domain name, and SHOULD have its IP address. Conformant clients SHOULD validate that the name or IP address of the cert matches the server that the client connected to.

I noted before in the patch adding SSL to epee that the cert generation function is missing steps to properly set the subject*Name fields of the generated certs.

Reply:
I do not want to leak a fq server name in the cert (though vtnerd added support for this after my patch). Ideally people would whitelist fingerprints of the certs they intend to accept, rather than rely on name matching, which relies on centralized signature "authorities".

Reply:
OK, I guess that makes sense.